src/collision.cpp

```diff
@@ -614,3 +489,212 @@ collisionMoveResult collisionMoveSimple(Environment *env, IGameDef *gamedef,
 
 	return result;
 }
+
+bool collisionCheckIntersection(Environment *env, IGameDef *gamedef,
+		const aabb3f &box_0, const v3f &pos_f, ActiveObject *self,
+		bool collideWithObjects)
+{
+	#define PROFILER_NAME(text) (s_env ? ("Server: " text) : ("Client: " text))
+	Map *map = &env->getMap();
+	ServerEnvironment *s_env = dynamic_cast<ServerEnvironment*>(env);
```

Comment:
Multiple issues:
 - `PROFILER_NAME` macro duplicated -> should only be defined once

 - `ServerEnvironment *s_env = dynamic_cast<ServerEnvironment*>(env);` duplicated and only used for the `PROFILER_NAME` macro -> should be moved there

- `Map *map = &env->getMap();` only exists for passing the map into `add_area_node_boxes` -> should at least be moved closer to `add_area_node_boxes` OR you could instead pass the Environment pointer into `add_area_node_boxes` and do it there